kratos/processes/apply_periodic_boundary_condition_process.h

```diff
@@ -0,0 +1,196 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//
+//
+
+#ifndef APPLY_PERIODIC_CONDITION_PROCESS_H
+#define APPLY_PERIODIC_CONDITION_PROCESS_H
+
+// System includes
+#include <string>
+#include <iostream>
+#include <chrono>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "geometries/point.h"
+#include "processes/process.h"
+#include "utilities/math_utils.h"
+#include "includes/kratos_parameters.h"
+#include "includes/dof.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+
+class ApplyPeriodicConditionProcess : public Process
+{
+
+  public:
+    /// Pointer definition of ApplyPeriodicConditionProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyPeriodicConditionProcess);
+
+    typedef Dof<double>*                                    DofPointerType;
+    typedef Dof<double>                                     DofType;
+    typedef ModelPart::VariableComponentType                VariableComponentType;
+    typedef KratosComponents<Variable<array_1d<double, 3>>> VectorVariableType;
+    typedef ProcessInfo                                     ProcessInfoType;
+    typedef ProcessInfo::Pointer                            ProcessInfoPointerType;
+    typedef unsigned int                                    IndexType;
```

Comment:
std::size_t

Reply:
Yes ... this was as I remember to avoid compilation warnings on comparison between signed and unsigned integer.

Reply:
NodeType::IndexType